### PR TITLE
Enable jakartaee for rest connector and rest handler

### PR DESCRIPTION
**Pull request by anjumfatima90**
for issue #11885 & #11886

### CODE REVIEW

**Comment by anjumfatima90** [comment]
#build

**Comment by anjumfatima90** [comment]
#build

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.webBundleSecurity-1.0.feature [hunk 1]
```diff
@@ -0,0 +1,16 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+
+symbolicName = io.openliberty.webBundleSecurity-1.0
```
Comment (tjwatson) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.webBundleSecurity-1.0.feature:3]: I  suggest we call this io.openliberty.webBundleSecurity.javaee-1.0
That distinguishes it more from the main webBundleSecurity feature.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.webBundleSecurity-1.0.feature [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+visibility = private
+
+-bundles= com.ibm.ws.webcontainer.security; start-phase:=APPLICATION_EARLY, \
+          com.ibm.ws.security.authentication.tai
+
+IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=com.ibm.wsspi.appserver.webBundleSecurity-1.0))",\
+  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=com.ibm.websphere.appserver.servlet-3.1)(osgi.identity=com.ibm.websphere.appserver.servlet-4.0)))"
```
Comment (tjwatson) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.webBundleSecurity-1.0.feature:10]: I think you need to also be enabled with servlet-3.0

File: dev/com.ibm.ws.webcontainer.servlet.3.1.factories/bnd.bnd [hunk 3]
```diff
@@ -108,3 +108,5 @@ instrument.disabled: true
     com.ibm.ws.javaee.version;version=latest,\
     com.ibm.ws.artifact;version=latest,\
     com.ibm.ws.artifact.overlay;version=latest
+    
+jakartaeeMe: true
```
Comment (dazavala) [inline dev/com.ibm.ws.webcontainer.servlet.3.1.factories/bnd.bnd:112]: I don't see where this transformed bundle is used in any of the committed sources.  Is this change necessary?

**Comment by anjumfatima90** [comment]
#build

**Comment by andrea-pichler** [comment]
No message review requested from L2

**Comment by anjumfatima90** [comment]
#build

**Comment by anjumfatima90** [comment]
#build

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restConnector.javaee-2.0.feature [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+-bundles= com.ibm.ws.jmx.connector.server.rest
+
+IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=com.ibm.websphere.appserver.restConnector-2.0))",\
+  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=com.ibm.websphere.appserver.servlet-3.0)(osgi.identity=com.ibm.websphere.appserver.servlet-3.1)(osgi.identity=com.ibm.websphere.appserver.servlet-4.0)))"
+
+IBM-Install-Policy: when-satisfied
+
+edition=core
+kind=ga
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restConnector.javaee-2.0.feature:14]: Please set this to noship to start with.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restConnector.javaee-2.0.feature [hunk 5]
```diff
@@ -0,0 +1,15 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+
+symbolicName = io.openliberty.restConnector.javaee-2.0
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restConnector.javaee-2.0.feature:3]: This name looks odd to me because there is no such feature as `javaee-2.0`. I suggest calling this `io.openliberty.restConnector2.0-servlet`.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restHandler.jakarta-1.0.feature [hunk 6]
```diff
@@ -0,0 +1,15 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+
+symbolicName = io.openliberty.restHandler.jakarta-1.0
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restHandler.jakarta-1.0.feature:3]: This name looks odd to me because there is no such feature as `jakarta-1.0`. I suggest calling this `io.openliberty.restHandler1.0-servlet5.0`.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restHandler.javaee-1.0.feature [hunk 7]
```diff
@@ -0,0 +1,15 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+
+symbolicName = io.openliberty.restHandler.javaee-1.0
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restHandler.javaee-1.0.feature:3]: similar to above

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restHandler.javaee-1.0.feature [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+-bundles= com.ibm.ws.rest.handler
+
+IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=com.ibm.websphere.appserver.restHandler-1.0))",\
+  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=com.ibm.websphere.appserver.servlet-3.0)(osgi.identity=com.ibm.websphere.appserver.servlet-3.1)(osgi.identity=com.ibm.websphere.appserver.servlet-4.0)))"
+
+IBM-Install-Policy: when-satisfied
+
+edition=core
+kind=ga
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restHandler.javaee-1.0.feature:14]: noship

File: dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.adminSecurity-2.0.feature [hunk 9]
```diff
@@ -0,0 +1,13 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+symbolicName=com.ibm.websphere.appserver.adminSecurity-2.0
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.adminSecurity-2.0.feature:2]: io.openliberty....

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.webBundleSecurity.jakarta-1.0.feature [hunk 10]
```diff
@@ -0,0 +1,16 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+
+symbolicName = io.openliberty.webBundleSecurity.jakarta-1.0
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.webBundleSecurity.jakarta-1.0.feature:3]: This name looks odd to me because there is no such feature as `jakarta-1.0`. I suggest calling this `io.openliberty.webBundleSecurity1.0-servlet5.0`.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.webBundleSecurity.javaee-1.0.feature [hunk 11]
```diff
@@ -0,0 +1,16 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+
+symbolicName = io.openliberty.webBundleSecurity.javaee-1.0
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.webBundleSecurity.javaee-1.0.feature:3]: This name looks odd to me because there is no such feature as `javaee-1.0`. I suggest calling this `io.openliberty.webBundleSecurity1.0-servlet3.0`.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restConnector.javaee-2.0.feature (hunk 5 above)
Comment (tjwatson) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restConnector.javaee-2.0.feature:3]: But server3.0 is odd because it is enabled for all servlet versions in javaee (3.0, 3.1, 4.0).  If this is to change I suggest changing it to `io.openliberty.restConnector.javaee`

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restHandler.jakarta-1.0.feature (hunk 6 above)
Comment (tjwatson) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restHandler.jakarta-1.0.feature:3]: This feature will get updated to also tolerate servlet-5.0+ when Jakarta 10 releases.  If it needs to be renamed then I suggest renaming to `io.openliberty.restHandler.jakarta`.  The version after the last `-` is intended to version the overall auto-feature for singleton toleration reasons.  But no auto-feature should be a singleton so there is no need to have the `-1.0` on the end of the name.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.webBundleSecurity.jakarta-1.0.feature (hunk 10 above)
Comment (tjwatson) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.webBundleSecurity.jakarta-1.0.feature:3]: If we need to rename I suggest `io.openliberty.restHandler.jakarta` as before or perhaps `io.openliberty.restHandler1.0.jakarta`

File: dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.adminSecurity-2.0.feature (hunk 9 above)
Comment (tjwatson) [reply dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.adminSecurity-2.0.feature:2]: This cannot happen because we need to tolerate across the version of the private feature in `restHandler-1.0`

If we really want io.openliberty used here then we should do a concerted effort to rename ALL of our private features to io.openliberty...

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restConnector.javaee-2.0.feature (hunk 5 above)
Comment (tevans78) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restConnector.javaee-2.0.feature:3]: OK, that is a little less confusing.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restHandler.jakarta-1.0.feature (hunk 6 above)
Comment (tevans78) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restHandler.jakarta-1.0.feature:3]: OK, that is a little less confusing.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.webBundleSecurity.jakarta-1.0.feature (hunk 10 above)
Comment (tevans78) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.webBundleSecurity.jakarta-1.0.feature:3]: Either of those are fine

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.webBundleSecurity.javaee-1.0.feature (hunk 11 above)
Comment (anjumfatima90) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.webBundleSecurity.javaee-1.0.feature]: Changed to io.openliberty.webBundleSecurity1.0.javaee. Thanks!

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.webBundleSecurity.jakarta-1.0.feature (hunk 10 above)
Comment (anjumfatima90) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.webBundleSecurity.jakarta-1.0.feature]: Changed filename to `io.openliberty.webBundleSecurity1.0.jakarta`. Thanks!

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restHandler.jakarta-1.0.feature (hunk 6 above)
Comment (anjumfatima90) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restHandler.jakarta-1.0.feature]: Changed filename to `io.openliberty.restHandler1.0.jakarta`. Thanks!

File: dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.adminSecurity-2.0.feature (hunk 9 above)
Comment (anjumfatima90) [reply dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.adminSecurity-2.0.feature:2]: Keeping it the same for now.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restConnector.javaee-2.0.feature (hunk 5 above)
Comment (anjumfatima90) [reply dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restConnector.javaee-2.0.feature]: Changed filename to `io.openliberty.restConnector2.0.javaee`. Thanks!

**Comment by anjumfatima90** [comment]
#run-libby-bot
